.gitlab/deploy/runner/kubernetes/deploy.sh

```diff
@@ -33,24 +33,63 @@ if [ -z "${runner_registration_token}" ]; then
   exit 1
 fi
 
-kubectl apply -f - <<EOF
-apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRoleBinding
-metadata:
-  name: gitlab-runner:cluster-admin
-subjects:
-- kind: ServiceAccount
-  name: default
-  namespace: ${namespace}
-roleRef:
-  kind: ClusterRole
-  name: cluster-admin
-  apiGroup: rbac.authorization.k8s.io
-EOF
-
 helm upgrade "${installation_name}" gitlab/gitlab-runner \
   --install \
+  --wait \
   --namespace "${namespace}" \
   --values .gitlab/deploy/runner/kubernetes/values.yaml \
   --set "runners.namespace=${namespace}" \
   --set "runnerRegistrationToken=${runner_registration_token}"
+
+# shellcheck disable=SC1004,SC2016
+kubectl patch "configmap/${installation_name}-gitlab-runner" \
+  --namespace "${namespace}" \
+  --patch '
+"data":
+  "entrypoint": |
+    #!/bin/bash
+    set -e
+    mkdir -p /home/gitlab-runner/.gitlab-runner/
+    cp /scripts/config.toml /home/gitlab-runner/.gitlab-runner/
+
+    # Register the runner
+    if [[ -f /secrets/accesskey && -f /secrets/secretkey ]]; then
+      export CACHE_S3_ACCESS_KEY=$(cat /secrets/accesskey)
+      export CACHE_S3_SECRET_KEY=$(cat /secrets/secretkey)
+    fi
+
+    if [[ -f /secrets/gcs-applicaton-credentials-file ]]; then
+      export GOOGLE_APPLICATION_CREDENTIALS="/secrets/gcs-applicaton-credentials-file"
+    else
+      if [[ -f /secrets/gcs-access-id && -f /secrets/gcs-private-key ]]; then
```

Comment:
This is a dump of the original config, I don't want to touch what I didn't write.

Reply:
I'll add a comment explaining it.

Reply:
Yes, please add a comment on what changed, so that we could better maintain it.
Silly idea: `kubectl get "configmap/${installation_name}-gitlab-runner" --output=jsonpath='{.data.entrypoint}'` → `patch` → `kubectl patch`.